airflow/utils/python_virtualenv.py

```diff
@@ -35,12 +37,35 @@ def _generate_virtualenv_cmd(tmp_dir: str, python_bin: str, system_site_packages
     return cmd
 
 
-def _generate_pip_install_cmd(tmp_dir: str, requirements: List[str]) -> Optional[List[str]]:
+def _generate_pip_install_cmd(tmp_dir: str,
+                              requirements: List[str],
+                              connection_id: Optional[str] = None
+                              ) -> Optional[List[str]]:
     if not requirements:
         return None
-    # direct path alleviates need to activate
-    cmd = [f'{tmp_dir}/bin/pip', 'install']
-    return cmd + requirements
+
+    if connection_id:
+        con: Connection = BaseHook.get_connection(connection_id)
+        user = con.login
+        schema = con.schema or 'http'
+        password = con.get_password()
+        port = con.port
+        host = con.host
+        if user:
+            index_url = f"{schema}://{user}:{password}@{host}:{port}/repository/python/simple"
+        else:
+            index_url = f"{schema}://{host}:{port}/repository/python/simple"
+        private_cmd = [f'{tmp_dir}/bin/pip',
+                       'install',
+                       f'--trusted-host', host,
```

Comment:
I will revert to @uranusjr  here. He is our local PyPI expert (and pypi committer) so I trust 100% in his statements here.